bio.go

```diff
@@ -137,30 +149,40 @@ func (b *writeBio) WriteTo(w io.Writer) (rv int64, err error) {
 
 func (self *writeBio) Disconnect(b *C.BIO) {
 	if loadWritePtr(b) == self {
-		writeBioMapping.Del(token(C.X_BIO_get_data(b)))
 		C.X_BIO_set_data(b, nil)
 	}
 }
 
-func (b *writeBio) MakeCBIO() *C.BIO {
+func (b *writeBio) makeCBIO() *C.BIO {
 	rv := C.X_BIO_new_write_bio()
-	token := writeBioMapping.Add(unsafe.Pointer(b))
-	C.X_BIO_set_data(rv, unsafe.Pointer(token))
+	C.X_BIO_set_data(rv, unsafe.Pointer(b))
```

Comment:
just a comment: this set data/get data logic with the pointers I do not entirely like. I have a feeling that it could have been simpler. (I do not have a solution, I am only concerned about possible problems here. This is not related to the patch, but more to the general design here.)